docs/sources/reference/components/loki/loki.enrich.md

```diff
@@ -0,0 +1,153 @@
+---
+aliases:
+- /docs/alloy/latest/reference/components/loki/loki.enrich/
+canonical: /docs/alloy/latest/reference/components/loki/loki.enrich/
+title: loki.enrich
+labels:
+  stage: experimental
+description: The loki.enrich component enriches logs with labels from service discovery.
+---
+
+# loki.enrich
+
+{{< docs/shared lookup="stability/experimental.md" source="alloy" version="<ALLOY_VERSION>" >}}
+
+The `loki.enrich` component enriches logs with additional labels from service discovery targets. It matches a label from incoming logs against a label from discovered targets, and copies specified labels from the matched target to the log entry.
+
+## Usage
+
+```alloy
+loki.enrich "LABEL" {
+  // List of targets from a discovery component
+  targets = DISCOVERY_COMPONENT.targets
+  
+  // Which label from discovered targets to match against
+  match_label = "LABEL"
+  
+  // Which label from incoming logs to match against
+  source_label = "LABEL"
+  
+  // List of labels to copy from discovered targets to logs
+  target_labels = ["LABEL", ...]
+  
+  // Where to send enriched logs
+  forward_to = [RECEIVER_LIST]
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`targets` | `[]discovery.Target` | List of targets from a discovery component. | | yes
+`target_match_label` | `string` | Which label from discovered targets to match against (e.g., "__inventory_consul_service"). | | yes
```

Comment:
Is it enough to use just one label? I'm not sure but it might be useful to be able to match multiple.

Reply:
I thought about it and I think this is probably edge case that we could work around with using relabeling to join some of the labels together first.

Reply:
Ok, let's not overcomplicate it for now.